frontend/src2/types/chart.types.ts

```diff
@@ -51,6 +51,7 @@ export type SeriesBar = Series & {
 
 export type BarChartConfig = AxisChartConfig & {
 	y_axis: YAxisBar
+	label_rotation?: number
```

Comment:
this should ideally be under the `x_axis` configuration, but `x_axis` is currently directly mapped to a `dimension`. It needs to be mapped to a new type `XAxis` similar to `YAxis` to add more `x_axis` options

I'll make that change first and also patch the old options to new options then merge this

Reply:
Got it, thanks for the feedback!